src/main/java/com/valuemotive/lemon/ParkingManager.java

```diff
@@ -1,20 +1,20 @@
-package com.lemon;
-
+package com.valuemotive.lemon;
 
 import java.util.List;
 import java.util.Optional;
 import java.util.Random;
 
-import com.lemon.Car;
-import com.lemon.ParkingSlot;
+import com.valuemotive.lemon.Car;
+import com.valuemotive.lemon.ParkingSlot;
 
-public class ParkingUtils {
+public class ParkingManager {
 
 	private static final Random random = new Random(System.currentTimeMillis());
 
 	public static int generateId() {
 		return Math.abs(random.nextInt());
 	}
+
 	public static final Optional<ParkingSlot> getFirstAvailableSlot(List<ParkingSlot> slotList) {
 		if (slotList == null) {
 			return Optional.empty();
```

Comment:
Move null check to call site i.e. where this function is called. This function should only accept defined lists